packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -37,46 +37,55 @@ type DepositedEvent = {type: 'Deposited'; ethersEvent?: Event} & HoldingUpdatedA
 type AssetTransferredEvent = {type: 'AssetTransferred'; ethersEvent: Event} & AssetTransferredArg;
 type ContractEvent = DepositedEvent | AssetTransferredEvent;
 
-// TODO: is it reasonable to assume that the ethAssetHolder address is defined as runtime configuration?
-/* eslint-disable no-process-env, @typescript-eslint/no-non-null-assertion */
-const ethAssetHolderAddress = makeAddress(
-  process.env.ETH_ASSET_HOLDER_ADDRESS || constants.AddressZero
-);
-const nitroAdjudicatorAddress = makeAddress(
-  process.env.NITRO_ADJUDICATOR_ADDRESS! || constants.AddressZero
-);
-/* eslint-enable no-process-env, @typescript-eslint/no-non-null-assertion */
-
-function isEthAssetHolder(address: Address): boolean {
-  return address === ethAssetHolderAddress;
-}
-
 const blockConfirmations = 5;
 
 export class ChainService implements ChainServiceInterface {
-  private readonly ethWallet: NonceManager;
-  private provider: providers.JsonRpcProvider;
-  private allowanceMode: AllowanceMode;
   private addressToObservable: Map<Address, Observable<ContractEvent>> = new Map();
   private addressToContract: Map<Address, Contract> = new Map();
   private channelToSubscription: Map<Bytes32, Subscription[]> = new Map();
-  private nitroAdjudicator: Contract;
 
   private transactionQueue = new PQueue({concurrency: 1});
 
-  constructor({provider, pk, allowanceMode, pollingInterval}: ChainServiceArgs) {
-    this.provider = new providers.JsonRpcProvider(provider);
-    this.allowanceMode = allowanceMode;
-    if (provider.includes('0.0.0.0') || provider.includes('localhost')) {
-      pollingInterval = pollingInterval ?? 50;
-    }
-    if (pollingInterval) this.provider.pollingInterval = pollingInterval;
-    this.ethWallet = new NonceManager(new Wallet(pk, new providers.JsonRpcProvider(provider)));
-    this.nitroAdjudicator = new Contract(
+  static async create({
+    provider,
+    pollingInterval,
+    pk,
+    allowanceMode,
+    nitroAdjudicatorAddress,
+    ethAssetHolderAddress,
+  }: ChainServiceArgs): Promise<ChainService> {
+    const jsonrpc = new providers.JsonRpcProvider(provider);
+
+    if (pollingInterval) jsonrpc.pollingInterval = pollingInterval;
+
+    if (provider.includes('0.0.0.0') || provider.includes('localhost'))
+      jsonrpc.pollingInterval = pollingInterval ?? 50;
```

Comment:
This might be more clear. This condition is satisfied when the provider is `0.0.0.0` or localhost AND polling interval has not been set in the line above.
```suggestion
    if ((provider.includes('0.0.0.0') || provider.includes('localhost')) && !pollingInterval )
      jsonrpc.pollingInterval = 50;
```